smaug/operators/padding_op.h

```diff
@@ -19,95 +20,77 @@ namespace smaug {
 template <typename Backend>
 class PaddingOp : public Operator {
    public:
-    PaddingOp(const std::string& name,
-              Workspace* workspace)
-            : Operator(name, OpType::Repeat, workspace){
+    PaddingOp(const std::string& name, Workspace* workspace)
+            : Operator(name, OpType::Padding, workspace) {
         inputs.resize(kNumInputs, nullptr);
         outputs.resize(kNumOutputs, nullptr);
     }
 
-    PaddingOp(const std::string& name,
-              Workspace* workspace,
-              int val)
-            : Operator(name, OpType::Repeat, workspace), padder(val){
-        inputs.resize(kNumInputs, nullptr);
-        outputs.resize(kNumOutputs, nullptr);
+    /**
+     * Set the paddingSize of the Tensor along each dimension.
+     * The paddingSize is orgainized as <dim1_forward, dim1_backward, ...
+     * ,dimk_backward>
+     */
+    void setPaddingSize(RepeatedField<google::protobuf::int32> const& val) {
+        paddingSize.assign(val.begin(), val.end());
     }
 
-    /** Set the number of padders of the Tensor along each dimension. */
-    void setPadder(const int& val) {
-        padder = val;
-        // set output size?
-    }
+    void setPaddingSize(std::vector<int> const& val) { paddingSize = val; }
 
-    int getPadder() { return padder; }
+    std::vector<int> getPaddingSize() const { return paddingSize; }
 
     void run() override {
-      Tensor* input = getInput(0);
-      Tensor* output = getOutput(0);
-      int ndims = input->ndims();
-      std::vector<int> inputDims = input->getShape().dims();
-      std::vector<int> outputDims = output->getShape().dims();
-      int total_dim = 1;
-      for (int i: outputDims){
-        total_dim *= i;
-      }
-      std::vector<float> vf(total_dim, 0);
-      output->fillData(vf.data(), vf.size());
-      /*
-      copyTensorRegion(Tensor* dest,
-                      Tensor* src,
-                      const std::vector<int>& destOrigin,
-                      const std::vector<int>& srcOrigin,
-                      const std::vector<int>& regionSize
-      */
-      std::vector<int> destOrigin;
-      if (input->getShape().getLayout() == DataLayout::NCHW){
-        destOrigin = std::vector<int>({0, 0, padder, padder});
-      }
-      else if(input->getShape().getLayout() == DataLayout::NHWC){
-        destOrigin = std::vector<int>({0, padder, padder, 0});
-      }
-      else{
-        assert(false && "Invalid padding data type!");
-      }
-      std::vector<int> srcOrigin = std::vector<int>({0, 0, 0, 0});
-      std::vector<int> regionSize = inputDims;
-      copyTensorRegion(output, input, destOrigin, srcOrigin, regionSize);
+        Tensor* input = getInput(0);
```

Comment:
Use the enums defined below (`kInput` and `kOutput`) to access all the input/output tensors, here and everywhere else in this file.